modal/_utils/function_utils.py

```diff
@@ -249,6 +249,12 @@ def get_globals(self) -> dict[str, Any]:
         from .._vendor.cloudpickle import _extract_code_globals
 
         func = self.raw_f
+        while True:
+            # Unwrap functions decorated using functools.wrapped (potentially multiple times)
+            try:
```

Comment:
This is a really contrived example and I hope no one will do this:

```python
import functools


def f():
    print("hi")


functools.wraps(f)(f)

print(f.__wrapped__ is f) # True
print(f.__wrapped__.__wrapped__ is f) # True
```

But just in case, we could guard against `func != func.__wrapped__` so we don't go into an infinite loop?